cuqi/distribution/_truncated_normal.py

```diff
@@ -0,0 +1,48 @@
+import numpy as np
+from scipy.special import erf
+from cuqi.distribution import Distribution
+
+class TruncatedNormal(Distribution):
+    """
+    Truncated Normal probability distribution. Generates instance of cuqi.distribution.TruncatedNormal.
+    
+    The variables of this distribution are iid.
+
+    
+    Parameters
+    ------------
+    mean: mean of distribution
+    std: standard deviation
+    a: lower bound of the distribution
+    b: upper bound of the distribution
+    """
+    def __init__(self, mean=None, std=None, a=-np.Inf, b=np.Inf, is_symmetric=False, **kwargs):
+        # Init from abstract distribution class
+        super().__init__(is_symmetric=is_symmetric, **kwargs)  
+
+        # Init specific to this distribution
+        self.mean = mean
+        self.std = std
+        self.a = a
+        self.b = b
+
+    def logpdf(self, x):
+        # the unnormalized logpdf
+        # check if x falls in the range between np.array a and b
+        if np.any(x < self.a) or np.any(x > self.b):
+            return -np.Inf
+        else:
+             return np.sum(-np.log(self.std*np.sqrt(2*np.pi))-0.5*((x-self.mean)/self.std)**2)
```

Comment:
One trick we did in lognormal distribution is that we sat up an underlying Gaussian and used its logpdf and gradient. This gives more robustness in general because we relay on a single implementation that is heavily tested. Another advantage I believe is that we get the chain rule for the gradient for free in this case (e.g. when lognormal is used for likelihood for example). You can consider using this trick here as well.

Reply:
That's an excellent suggestion and following this strategy it will become way easier to adapt a truncated version of `cuqi.distribution.Gaussian` (instead of the iid `cuqi.distribution.Normal`)

Reply:
👍